Lib/test/test_capi/test_opt.py

```diff
@@ -842,6 +851,24 @@ def testfunc(n):
         # We'll also need to verify that propagation actually occurs.
         self.assertIn("_BINARY_OP_MULTIPLY_FLOAT", uops)
 
+    def test_add_unicode_propagation(self):
+        def testfunc(n):
+            a = ""
+            for _ in range(n):
+                a + a
+                a + a
+                a + a
+                a + a
```

Comment:
Sooner or later we'll have a peephole pass that will delete the generated code for this completely. Is there a reason you're not doing something like `a = a + a`?

Reply:
That transforms it to `BINARY_OP_INPLACE_ADD_UNICODE` sadly. Which is an unsupported opcode, so the trace ends there.